skyportal/handlers/api/annotation.py

```diff
@@ -122,30 +187,68 @@ def post(self):
             )
 
         author = self.associated_user_object
-        annotation = Annotation(
-            data=annotation_data,
-            obj_id=obj_id,
-            origin=origin,
-            author=author,
-            groups=groups,
-        )
+        if associated_resource_type.lower() in (
+            "object",
+            "objects",
+            "source",
+            "sources",
+        ):
+            if obj_id is None:
+                return self.error("Missing required field `obj_id`")
+            annotation = Annotation(
+                data=annotation_data,
+                obj_id=obj_id,
+                origin=origin,
+                author=author,
+                groups=groups,
+            )
+            annotation_resource_id_str = str(annotation.obj_id)
+        else:
+            return self.error(f'Unknown resource type "{associated_resource_type}".')
+
+        if annotation_resource_id_str != resource_id:
+            return self.error(
+                f'Annotation resource ID does not match resource ID given in path ({resource_id})'
+            )
 
         DBSession().add(annotation)
         self.verify_and_commit()
-        self.push_all(
-            action='skyportal/REFRESH_SOURCE',
-            payload={'obj_key': annotation.obj.internal_key},
-        )
+
+        if obj_id is not None:
```

Comment:
I think this is the correct check; you want to check whether it was specified (not None), and then do some checks to see if it is the right value.  On integers, you cannot simply do `if obj_id` because it might be zero.